api/src/test/java/org/openmrs/api/EncounterServiceTest.java

```diff
@@ -98,7 +97,235 @@
 	public void runBeforeEachTest() throws Exception {
 		executeDataSet(ENC_INITIAL_DATA_XML);
 	}
-	
+
+	@Test
+	@Verifies(value = "should save encounter correctly when a leaf child obs updated without recreating an entire obs hierarchy", method = "saveEncounter(Encounter)")
+	public void saveEncounter_shouldNotCreateACopyOfParentObsButEditChildObs(){
+		ConceptService cs = Context.getConceptService();
+		EncounterService es = Context.getEncounterService();
+		ObsService os = Context.getObsService();
+
+		//create an obs
+		Obs o = createObs(cs.getConcept(3),"original obs value text");
+		//create a second obs
+		Obs o2 = createObs(cs.getConcept(3),"second obs value text");
+		//create a parent obs
+		Obs oParent =createObs(cs.getConcept(23), null); //in the concept set table as a set
+
+		//add o and o2 to the parent obs
+		oParent.addGroupMember(o2);
+		oParent.addGroupMember(o);
```

Comment:
When you say leaf, I was assume one of the  obs should not be an immediate child but i see you're adding both as children of the same parent, i would expect a child-parent-grandparent tree